### PR TITLE
Refs [TASK][35898] Create UI Search Screen

### PR DESCRIPTION
**Pull request by datvt-2504**
## Related Tickets
- [#Task 35898: Create UI Search Screen](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/dmealfoodapp/ui/search/SearchFragment.kt [hunk 1]
```diff
@@ -2,10 +2,15 @@ package com.sunasterisk.dmealfoodapp.ui.search
 
 import com.sunasterisk.dmealfoodapp.R
 import com.sunasterisk.dmealfoodapp.base.BaseFragment
+import com.sunasterisk.dmealfoodapp.utils.gone
+import com.sunasterisk.dmealfoodapp.utils.show
+import kotlinx.android.synthetic.main.tool_bar_screen_child.*
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/dmealfoodapp/ui/search/SearchFragment.kt]: sao anh không dùng viewBinding thay cho kotlin synthetic 
https://developer.android.com/topic/libraries/view-binding

File: app/src/main/res/layout/fragment_search.xml [hunk 2]
```diff
@@ -1,12 +1,45 @@
...
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:background="@android:color/white"
     tools:context=".ui.search.SearchFragment">
 
-    <TextView
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
+    <include
+        android:id="@+id/include"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
        android:id="@+id/layoutToolBar"
```

File: app/src/main/res/layout/fragment_search.xml [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
+        layout="@layout/tool_bar_screen_child"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
        layout="@layout/layout_tool_bar_screen_child"
```

File: app/src/main/java/com/sunasterisk/dmealfoodapp/ui/search/SearchFragment.kt [hunk 4]
```diff
@@ -2,10 +2,15 @@ package com.sunasterisk.dmealfoodapp.ui.search
...
 
 class SearchFragment : BaseFragment(R.layout.fragment_search) {
 
     override fun onCreatedView() {
+        editTextSearch.show()
+        textTitleScreen.gone()
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/dmealfoodapp/ui/search/SearchFragment.kt]:
```suggestion
        textSearch.show()
        textTitle.gone()
```
edit text nếu để tên dài quá em có thể chỉ để mỗi text cũng đc

File: app/src/main/res/layout/fragment_search.xml [hunk 5]
```diff
@@ -1,12 +1,45 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
-</FrameLayout>
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerViewSearchMeal"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
        android:id="@+id/recyclerMeals"
```

File: app/src/main/res/layout/tool_bar_screen_child.xml [hunk 6]
```diff
@@ -32,6 +32,24 @@
         app:layout_constraintStart_toEndOf="@+id/buttonBack"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <EditText
+        android:id="@+id/editTextSearch"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/tool_bar_screen_child.xml:36]:
```suggestion
        android:id="@+id/textSearch"
```
để nguyên editTextSearch cũng đc

File: app/src/main/java/com/sunasterisk/dmealfoodapp/ui/search/SearchFragment.kt (hunk 1 above)
Comment (datvt-2504) [reply app/src/main/java/com/sunasterisk/dmealfoodapp/ui/search/SearchFragment.kt]: lúc đầu a chưa nghiên cứu đến cái này nên dùng find view bằng cách sử dụng plugin kotlin extensions, cái này a sẽ tìm hiểu để dùng vào project sau

**Review by tuannm-1276 — APPROVED** [review]

**Pull request merged**